python/integration_tests/helpers.py

```diff
@@ -90,29 +102,64 @@ def serialize_task_activation(i, args: list, kwargs: dict) -> bytes:
     return pending_task_payload
 
 
-def send_messages_to_kafka(topic_name: str, num_messages: int) -> None:
+def send_generic_messages_to_topic(topic_name: str, num_messages: int) -> None:
     """
     Send num_messages to kafka topic using unique task names.
     """
     try:
-        producer = Producer(
-            {
-                "bootstrap.servers": "127.0.0.1:9092",
-                "broker.address.family": "v4",
-            }
-        )
+        producer = Producer(TEST_PRODUCER_CONFIG)
 
         for i in range(num_messages):
             task_message = serialize_task_activation(i, ["foobar"], {})
             producer.produce(topic_name, task_message)
 
         producer.poll(5)  # trigger delivery reports
         producer.flush()
-        print(f"Sent {num_messages} messages to kafka topic {topic_name}")
+        print(f"Sent {num_messages} generic messages to kafka topic {topic_name}")
     except Exception as e:
         raise Exception(f"Failed to send messages to kafka: {e}")
 
 
+def send_custom_messages_to_topic(topic_name: str, custom_messages: list[TaskActivation]) -> None:
+    """
+    Send num_messages to kafka topic using unique task names.
+    """
+    try:
+        producer = Producer(TEST_PRODUCER_CONFIG)
+
+        for message in custom_messages:
+            task_message = message.SerializeToString()
+            producer.produce(topic_name, task_message)
+
+        producer.poll(5)  # trigger delivery reports
+        producer.flush()
```

Comment:
I don't think you need the flush here, my understanding is that `flush` is a wrapper for `poll(0)`.